### ISSUE TITLE
How to visualiaze loss values getting decreased during training?

### ISSUE THREAD

**Issue by chowkamlee81** [tianweiy/CenterPoint#3]
Is tensorboard is used to visualize since all are log files.

**Comment by tianweiy** [comment]
No, it is not used.

**Comment by tianweiy** [comment]
I just use the raw log to see the changes of the loss

**Comment by tianweiy** [comment]
Though you should be able to use tensorboard if you want. There is a tensorboard logger hook in ```det3d/torchie/trainer/hooks/logger/tensorboard.py```. You need to adapt this file according to ```det3d/torchie/trainer/hooks/logger/text.py``` and change the config files.

**Comment by chowkamlee81** [comment]
Thanks Ok

**Comment by YazanMurhij995** [comment]
I want to view the model as a graph in tensorboard using this command:
writer = SummaryWriter()
writer.add_graph(model, input)
where I should insert this command ?

**Comment by tianweiy** [comment]
Hi, currently we can't use tensorboard to view the model as the input / output doesn't follow the predefined format (e.g. iterable). You can view the module through print. I remember @muzi2045  gets some graph visualization and it will be great if you(@muzi2045 ) can explain some simple way to do this.

**Comment by chowkamlee81** [comment]
@YazanMurhij995 
Kindly use onnx. Convert pytorch model to onnx format. You can easily visualize onnx file because already software s are there to view onnx files

**Comment by YazanMurhij995** [comment]
I tried but I am facing the same problem. it is not clear how to specify an input to the model! Can you give an example  of the input please.
It is something  like Dict(voxels, num_points, num_voxels, coords, [grid_size])

**Comment by YazanMurhij995** [comment]
@chowkamlee81 , Have you managed to convert the model to onnx? it seems that the input format is not supported. Either the dict or np.ndarray should be changed.
Can you tell me, please?

**Comment by Qjizhi** [comment]
> @chowkamlee81 , Have you managed to convert the model to onnx? it seems that the input format is not supported. Either the dict or np.ndarray should be changed.
> Can you tell me, please?

@YazanMurhij995 Do you have any idea yet?

**Comment by ChidanandKumarKS** [comment]
[loss_analysis.txt](https://github.com/tianweiy/CenterPoint/files/5702139/loss_analysis.txt)
Attaching the code to visulaize loss values over a sequence of time. 
@YazanMurhij995 , @tianweiy 
Hope this solves analyzing loss visualization issues

**Comment by YoushaaMurhij** [comment]
@Qjizhi you can try to export some parts of the model (i,e neck or head) only! as the Dict format is not supported yet

**Comment by Qjizhi** [comment]
> @Qjizhi you can try to export some parts of the model (i,e neck or head) only! as the Dict format is not supported yet

Hi @YoushaaMurhij, thanks for your reply, could you please share some open-sourced code?

**Comment by YoushaaMurhij** [comment]
Here are some instructions:
https://pytorch.org/docs/stable/onnx.html
you can apply it [here](https://github.com/tianweiy/CenterPoint/blob/604585ac6257a6b5e3cfb531cb36adb1034c2d4c/det3d/models/detectors/point_pillars.py#L29) (for example)

**Comment by xavidzo** [comment]
> @Qjizhi you can try to export some parts of the model (i,e neck or head) only! as the Dict format is not supported yet

@YoushaaMurhij  but if you change the inputs of the forward method of the reader, neck, and backbone to be lists instead of dictionaries, then the whole model could be exported easily to onnx? Can you answer please

**Comment by YoushaaMurhij** [comment]
@xavidzo Yes, It should be converted easily without problems. I didn't try that!

**Comment by xavidzo** [comment]
> @xavidzo Yes, It should be converted easily without problems. I didn't try that!

Ok, thank you. but if the output of the model, i.e. the output of the head is also a dictionary, would this work for onnx or should I also change the output to be a list as well?

**Comment by YoushaaMurhij** [comment]
I think it should work for the head, too. The problem with be only with the input part here.
